common/src/main/java/com/wynntils/core/features/overlays/Overlay.java

```diff
@@ -6,35 +6,54 @@
 
 import com.mojang.blaze3d.platform.Window;
 import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.features.overlays.sizes.FixedOverlaySize;
+import com.wynntils.core.features.overlays.sizes.OverlaySize;
+import com.wynntils.utils.objects.Pair;
 
 public abstract class Overlay {
     protected OverlayPosition position;
 
-    protected float width;
-    protected float height;
+    protected OverlaySize size;
 
     public Overlay(OverlayPosition position, float width, float height) {
         this.position = position;
-        this.width = width;
-        this.height = height;
+        this.size = new FixedOverlaySize(width, height);
     }
 
-    public abstract void render(
-            OverlayPosition overlayPosition, PoseStack poseStack, float partialTicks, Window window);
-
-    public OverlayPosition getPosition() {
-        return position;
+    public Overlay(OverlayPosition position, OverlaySize size) {
+        this.position = position;
+        this.size = size;
     }
 
-    public void setPosition(OverlayPosition position) {
-        this.position = position;
+    public abstract void render(PoseStack poseStack, float partialTicks, Window window);
+
+    public float getWidth() {
+        return this.size.getWidth();
     }
 
     public float getHeight() {
-        return height;
+        return this.size.getHeight();
     }
 
-    public float getWidth() {
-        return width;
+    // Return the X where the overlay should be rendered
+    public int getRenderX() {
+        final Pair<Coordinate, Coordinate> section = OverlayManager.getSection(this.position.getAnchorSection());
+        return switch (this.position.getHorizontalAlignment()) {
+            case Left -> section.a.x() + this.position.getHorizontalOffset();
+            case Center -> (int) (section.a.x() + section.b.x() - this.getWidth()) / 2
+                    + this.position.getHorizontalOffset();
+            case Right -> (int) (section.b.x() - this.position.getHorizontalOffset() - this.getWidth());
```

Comment:
The horizontal offset is added in the Left and Center cases, but subtracted in the Right case. It might be better to just have it always added. Same comment applies for vertical offset.

Reply:
Why? If, let's say, my vertical alignment is aligned to bottom, any positive offset should move the overlay further from the bottom. Same with horizontal right.

Reply:
It's inconsistent with Left and Center, and would make it harder to use imo.

Reply:
Yeah, sure, I don't mind either way. I will open a PR in a bit.